x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/converters/common_params_camel_to_snake.test.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getBaseRuleParams } from '../../../../rule_schema/mocks';
+import { commonParamsCamelToSnake } from './common_params_camel_to_snake';
+
+describe('commonParamsCamelToSnake', () => {
+  test('should convert rule_source params to snake case', () => {
+    const transformedParams = commonParamsCamelToSnake({
+      ...getBaseRuleParams(),
+      ruleSource: {
+        type: 'external',
+        isCustomized: false,
+      },
+    });
+    expect(transformedParams).toEqual(
+      expect.objectContaining({
```

Comment:
I'm a bit opposed to creating snapshot tests for the whole function output because snapshots have proven to be too flaky without providing any extra quality assurance.

Anyway, this test existed before, and my goal here was not to provide additional test coverage of the existing code. During the refactoring, I transferred the already existing tests to new locations and added new tests only for the newly introduced functionality, like the `isCustomized` calculation.